packages/gatsby-cli/src/reporter/loggers/ink/cli.js

```diff
@@ -0,0 +1,71 @@
+import React from "react"
+import { Box, Static } from "ink"
+import { isCI } from "ci-info"
+
+import Spinner from "../ink/components/spinner"
+import ProgressBar from "../ink/components/progress-bar"
+
+import { Message } from "../ink/components/messages"
+import Error from "./components/error"
+import Develop from "../ink/components/develop"
+
+const showProgress = process.stdout.isTTY && !isCI
+
+class CLI extends React.Component {
```

Comment:
I was just wondering because Develop.js ink component was rewritten to a functional component so that's why I was asking :joy:. Also unsure if Functional components break older react versions. We had this bug before:
https://github.com/gatsbyjs/gatsby/issues/14911#issuecomment-513172124

Reply:
Yeah, probably I discarded this in one of master merges (which are tricky given, that I moved file around). I'll take a look in making sure changes here are as minimal as possible vs master

Reply:
Refactored this to a functional component!

Reply:
Never mind, refactored back to a class component because we need to use `componentDidCatch` 

Also fixed some broken telemetry code in the process!